engine/gamesys/src/gamesys/components/comp_sprite.cpp

```diff
@@ -857,16 +857,25 @@ namespace dmGameSystem
                 world->m_RenderObjects.SetSize(0);
                 break;
             case dmRender::RENDER_LIST_OPERATION_END:
-                dmGraphics::SetVertexBufferData(world->m_VertexBuffer, sizeof(SpriteVertex) * (world->m_VertexBufferWritePtr - world->m_VertexBufferData),
-                                                world->m_VertexBufferData, dmGraphics::BUFFER_USAGE_STATIC_DRAW);
+                {
+                    uint32_t vertex_size = sizeof(SpriteVertex) * (world->m_VertexBufferWritePtr - world->m_VertexBufferData);
+                    if (vertex_size)
+                    {
+                        dmGraphics::SetVertexBufferData(world->m_VertexBuffer, vertex_size,
+                                                        world->m_VertexBufferData, dmGraphics::BUFFER_USAGE_DYNAMIC_DRAW);
 
-                DM_COUNTER("SpriteVertexBuffer", (world->m_VertexBufferWritePtr - world->m_VertexBufferData) * sizeof(SpriteVertex));
+                        DM_COUNTER("SpriteVertexBuffer", vertex_size);
+                    }
+                }
 
                 if (world->m_UseGeometries)
                 {
                     uint32_t index_size = (world->m_IndexBufferWritePtr - world->m_IndexBufferData);
-                    dmGraphics::SetIndexBufferData(world->m_IndexBuffer, index_size, world->m_IndexBufferData, dmGraphics::BUFFER_USAGE_STATIC_DRAW);
-                    DM_COUNTER("SpriteIndexBuffer", index_size);
+                    if (index_size)
+                    {
+                        dmGraphics::SetIndexBufferData(world->m_IndexBuffer, index_size, world->m_IndexBufferData, dmGraphics::BUFFER_USAGE_DYNAMIC_DRAW);
```

Comment:
For some reason, this call take an extremely large amount of time on WebGL 2 when given an empty buffer.

Reply:
Huh, that's weird!